src/Arrays/MPIStateArrays.jl

```diff
@@ -515,12 +515,27 @@ function LinearAlgebra.norm(
     p::Real = 2,
     weighted::Bool = true;
     dims = :,
+    realdata = Q.realdata,
 )
     if weighted && ~isempty(Q.weights) && isfinite(p)
         W = @view Q.weights[:, :, Q.realelems]
-        locnorm = weighted_norm_impl(Q.realdata, W, Val(p), dims)
+
+        # Check that the arrays are the right dimensionality
+        @assert ndims(realdata) in (2, 3)
+
+        # Reshape to 3 dimensions is necessary
+        if ndims(realdata) == 2
+            realdata =
+                reshape(realdata, size(realdata, 1), 1, size(realdata, 2))
+        end
```

Comment:
The other option here is to force the user to send in the data correctly shaped, but then they cannot use the `vars` pull-out from the `MPIStateArray`

Reply:
@jkozdon looks like one of the tests for this failed, can you take a look?

https://github.com/CliMA/ClimateMachine.jl/runs/831119353#step:7:235

Reply:
> @jkozdon looks like one of the tests for this failed, can you take a look?

Forgot to test in a parallel. Pushed a fix.